share/ipld/namespace_data.go

```diff
@@ -275,6 +278,62 @@ func (n *NamespaceData) CollectLeavesByNamespace(
 	}
 }
 
+func (n *NamespaceData) traverseLinks(j job, links []*ipld.Link) []job {
+	if j.isAbsent {
+		return n.collectAbsenceProofs(j, links)
+	}
+	return n.collectNDWithProofs(j, links)
+}
+
+func (n *NamespaceData) collectAbsenceProofs(j job, links []*ipld.Link) []job {
+	leftLink := links[0].Cid
+	rightLink := links[1].Cid
+	// traverse to the left node, while collecting right node as proof
+	n.addProof(right, rightLink, j.depth)
+	return []job{j.next(left, leftLink, j.isAbsent)}
+}
+
+func (n *NamespaceData) collectNDWithProofs(j job, links []*ipld.Link) []job {
+	leftCid := links[0].Cid
+	rightCid := links[1].Cid
+	leftLink := NamespacedSha256FromCID(leftCid)
+	rightLink := NamespacedSha256FromCID(rightCid)
+
+	var nextJobs []job
+	// check if target namespace is outside of boundaries of both links
+	if NamespaceIsOutsideRange(leftLink, rightLink, n.nID) {
+		if j.depth == 0 {
+			// target namespace is not present in the root namespaces range
+			n.isAbsentNamespace.Store(true)
+			return nil
```

Comment:
Okay we discussed this sync and the only chance this will happen is if somehow a row root that is either < or > namespace hits this path (unlikely) but if we are going to handle it, then there should be some visibility into this error occurring (which there currently isn't)

Reply:
The internal if check is impossible to happen with current pre-validation check, so I'll remove it as dead code.